### ISSUE TITLE
Leverage common container code for NativeAOT EventPipe 

### ISSUE THREAD

**Issue by LakshanF** [dotnet/runtime#81002]
Pending PR #80382 once checked in as is, leverages its own skeleton container code for its use. This is a temporary step until the pending PR #78852, which provides a common container source code in the shared code, is merged into the tree. Once that is done, 
 - Pending PR #80382 should move to the common code.
 - Explore building the common EventPipe source code, which is written in C (C99) without the C++ flag
